src/lightning_app/utilities/packaging/cloud_compute.py

```diff
@@ -100,12 +106,21 @@ def __post_init__(self):
         if self._internal_id is None:
             self._internal_id = "default" if self.name == "default" else uuid4().hex[:7]
 
-    def to_dict(self):
+    def to_dict(self) -> dict:
+        _verify_mount_root_dirs_are_unique(self.mount)
         return {"type": __CLOUD_COMPUTE_IDENTIFIER__, **asdict(self)}
 
     @classmethod
-    def from_dict(cls, d):
+    def from_dict(cls, d: dict) -> "CloudCompute":
         assert d.pop("type") == __CLOUD_COMPUTE_IDENTIFIER__
+        mounts = d.pop("mount", None)
+        if isinstance(mounts, dict):
+            d["mount"] = Mount(**mounts)
+        elif isinstance(mounts, (list, tuple, set)):
```

Comment:
why are we handling tuple and set if the type annotation was list?

Reply:
because it's not unreasonable for someone to pass a tuple or set of objects instead of a list? Usually for user facing APIs I try to handle any set of reasonable argument types instead of erroring on them. ya know... [duck typing and all that](https://towardsdatascience.com/duck-typing-python-7aeac97e11f8)

Is this not common practice in the lightning codebase?

Reply:
I'm not saying we shouldn't. I was pointing out that the implementation and type annotations seem to disagree.